testsuites/basic/timesync.py

```diff
@@ -0,0 +1,69 @@
+import re
+
+from assertpy import assert_that
+
+from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Windows
+from lisa.testsuite import simple_requirement
+from lisa.tools import Cat, Dmesg
+from lisa.util import get_matched_str
+
+
+@TestSuiteMetadata(
+    area="time",
+    category="functional",
+    description="""
+    This test suite is related with time sync.
+    """,
+    tags=["timesync"],
+)
+class TimeSync(TestSuite):
+    ptp_registered_dmesg_pattern = re.compile(r"PTP clock support registered", re.M)
+    hyperv_ptp_udev_rule_pattern = re.compile(r"ptp_hyperv", re.M)
+    chrony_path = ["/etc/chrony.conf", "/etc/chrony/chrony.conf"]
+
+    @TestCaseMetadata(
+        description="""
+        https://docs.microsoft.com/en-us/azure/virtual-machines/linux/time-sync#check-for-ptp-clock-source # noqa: E501
+        This test is to check -
+            1. PTP time source is available on Azure guests (newer versions of Linux).
+            2. PTP device name is hyperv.
+            3. When AN enabled, muiltple PTP devices will be available,
+             the names of ptp are changeable, create the symlink /dev/ptp_hyperv to whichever
+             /dev/ptp entry corresponds to the Azure host.
+            4. Chrony should be configured to use this symlink instead of
+             /dev/ptp0 or /dev/ptp1.
+        """,
+        priority=2,
+        requirement=simple_requirement(unsupported_os=[Windows]),
+    )
+    def timesync_validate_ptp(self, environment: Environment, node: Node) -> None:
+        # 1. PTP time source is available on Azure guests (newer versions of Linux).
```

Comment:
Don't we gotta have a distro allow-list, so that the test doesn't fail needlessly (we skip)?

Reply:
All endorsed distros should have it, and other derived distros shouldn't remove it in Azure. The test case can add a supported_platform requirement to avoid to run on other platforms. But for ready platform, the platform is unknown, it's not easy to check that. So if there is a platform other than Azure, Hyper-V, it can avoid to pick up this test case by modifying the test case filters in runbook file.

Reply:
But it's kernel version dependent, no? AFAIK Azure still endorses old, LTS kernels. Are all of those safe, is my question.

Reply:
Yes, if a feature is not supported in some distro as expected, the case should throw a skip exception to get it skipped with reason. It also needs an allowed list.

Reply:
we should have a allow-list, but I don't have one.